### PR TITLE
Add SMMA, replace RSI with original study

### PR DESCRIPTION
**Pull request by wuhkuh**
Part 1/2 of #759 

I'm going to split the pull request as this one is backwards compatible. The next part is not - it will replace the current RSI function with the one from the original study.

This means the patch version can be incremented for this specific PR, instead of a semver-minor.
Tests are included, however, it looked like the test suite is broken for this repository.
Anyway, I just added them in the same style as the tests for the other indicators.

The pre-calculated results are taken from my [own repository](https://github.com/wuhkuh/talib), which is documented and tested thoroughly. Sources used to determine the correct algorithm are documented, as well.

### CODE REVIEW

**Comment by askmike** [comment]
Thanks a lot! Great stuff, let me know when you have both parts ready and I'll go over them.

**Comment by wuhkuh** [comment]
Part 2/2 of #759

RSI has been replaced with the original study. Once again, tests are included and verified.
I haven't changed the interpretation of the RSI value, located under [strategies/RSI.js](https://github.com/askmike/gekko/tree/stable/strategies/RSI.js).

Keep in mind that this ~can~ will break the setup of the users.

I'd suggest doing a minor increment when implementing this, for this reason.

I assumed you'd want both parts together, so I put them together in this PR.
If not, I can split the pull request and send them in separately.

**Comment by askmike** [comment]
This PR looks great, thanks a lot for the time you put in :)

> Keep in mind that this ~can~ will break the setup of the users.

What exactly will this break besides calculating different results?

**Comment by wuhkuh** [comment]
> What exactly will this break besides calculating different results?

This is exactly it. I used the term break just to clarify that this change isn't a simple fix. I should've worded it differently, you're right.

Users of Gekko are probably used to working with the current results and I can imagine they wouldn't be too happy if we lower their revenue by implementing this change.

Assuming you're using Semver, if this version were to be 1.0.0 or higher, this change would classify as a semver-major - because we will change users results.
However, we aren't past 1.0.0 - so I'd suggest merging this whenever you're ready for 0.6.0.

**Comment by askmike** [comment]
> so I'd suggest merging this whenever you're ready for 0.6.0.

Great, I will make it part of the release described in #911 and merge it into `develop` (the nightly branch now).

------

> Assuming you're using Semver, if this version were to be 1.0.0 or higher, this change would classify as a semver-major - because we will change users results.

Semver describes versioning according to API changes, but since Gekko is not a library that exposes a single API I found semver hard to follow:

- Gekko (CLI) can be configured using a config, the form of which has changed quite drastically.
- Gekko (UI) can also be configured (hosts, ports, etc), this config format has also changed.
- Gekko writes files to disk, very rarely I had to update this format (causing the need for users to migrate data)
- Gekko exposes a REST API (which is undergoing a lot of changes).
- The functionality available to strats (the "strat API") (which this PR does not change) changes sometimes.
- Functionality changes (new exchanges, new markets, new functionality).

For that reason I found it impossible to follow strict semver since all the above can be argued to be "APIs", and a lot of them have had breaking changes in the past. Also I am not comfortable labeling this v1+.
